code/datums/datumvars.dm

```diff
@@ -387,53 +422,47 @@ body
 		#endif
 */
 	else if (isfile(value))
-		html += "[html_encode(name)] = <span class='value'>'[value]'</span>"
+		item = "[VV_HTML_ENCODE(name)] = <span class='value'>'[value]'</span>"
 
-	else if (istype(value, /client))
-		var/client/C = value
-		html += "<a href='?_src_=vars;Vars=\ref[value]'>[html_encode(name)] \ref[value]</a> = [C] [C.type]"
+	//else if (istype(value, /client))
+	//	var/client/C = value
+	//	item = "<a href='?_src_=vars;Vars=\ref[value]'>[VV_HTML_ENCODE(name)] \ref[value]</a> = [C] [C.type]"
 
 	else if (istype(value, /datum))
 		var/datum/D = value
-		html += "<a href='?_src_=vars;Vars=\ref[value]'>[html_encode(name)] \ref[value]</a> = [D.type]"
+		if ("[D]" != "[D.type]") //if the thing as a name var, lets use it.
+			item = "<a href='?_src_=vars;Vars=\ref[value]'>[VV_HTML_ENCODE(name)] \ref[value]</a> = [D] [D.type]"
+		else
+			item = "<a href='?_src_=vars;Vars=\ref[value]'>[VV_HTML_ENCODE(name)] \ref[value]</a> = [D.type]"
 
 	else if (istype(value, /list))
 		var/list/L = value
-		html += "[html_encode(name)] = /list ([L.len])"
-
-		if (L.len > 0 && !(name == "underlays" || name == "overlays" || name == "vars" || L.len > 500))
-			name = "[name]"	//Needs to be a string or it will go out of bounds in the internal_byond_list_vars array
-			html += "<ul>"
-			var/index = 1
-			for(var/entry in L)
-				var/state = "INDEX"
-				var/val = null
-				if(isnum(entry) || internal_byond_list_vars[name])
-					state = "INDEX"
-				else
-					val = L[entry]
-					if(!isnull(val))
-						state = "ASSOC"
-					if(isnull(L[index]))
-						state = "ASSOC"
-
-				switch(state)
-					if("INDEX")
-						html += debug_variable(index, L[index], level + 1)
-					if("ASSOC")
-						html += debug_variable(entry, val, level + 1)
-				index++
-			html += "</ul>"
+		var/list/items = list()
+
+		if (L.len > 0 && !(name == "underlays" || name == "overlays" || L.len > 500))

```

Comment:
I'd make it even lower, really can't think of where i wouldn't want list of size > 100 open in another window.